### DISCUSSION TITLE
Suddenly SPAM comes thru

### DISCUSSION THREAD

**Discussion by pixelman** [solspace/craft-freeform#346] [Questions & Answers] [answered]
The contact form of our site now suddenly lets spam through. 
It used to be stopped but now we get a lot of spam mails.

Is there any way how we can prevent this or manage this?
```
We use 
freeform pro 3.12.13
craft 3.7.33
```

<img width="1085" alt="image" src="https://user-images.githubusercontent.com/221232/168142734-49b83cf2-e471-4a36-9309-487e0382c484.png">

<img width="1068" alt="image" src="https://user-images.githubusercontent.com/221232/168144116-2e6bb401-81e5-421e-99c3-9f9b35e5cf65.png">

.

**Comment by kjmartens** [comment, marked as answer]
While Freeform comes with a [wide variety of spam protection](https://docs.solspace.com/craft/freeform/v3/overview/spam-protection.html) offerings, please keep in mind that no spam protection measure is 100% fool-proof, and if your site gets targetted by a spammer, you may need additional measures in place.

Here are the current spam protection options available, ordered from least to most effective (theoretically):

- Freeform's Honeypot
- Freeform's Honeypot with Javascript Enhancement enabled
- reCAPTCHA v2 Checkbox
- reCAPTCHA v2 Invisible (Pro only)
- reCAPTCHA v3 (Pro only)

Since you're currently just using the Freeform Honeypot feature, you may wish to switch to a reCAPTCHA offering, as you'll be more likely to have success. But please be aware that even reCAPTCHA can be circumvented.

Something else you can do is watch out for common trends and patterns in your spam emails. Making use of the Keyword and Email blocking features, you can block all emails with `*.ru` for example, or block keywords with `sex*`, `viagra`, `*href*`, etc. After monitoring and adjusting for a couple of weeks, you'll likely be able to alleviate the vast majority of spam this way. Just be careful not to block keywords that might actually be legitimate under some circumstances (unless you are certain to visit your Spam Folder regularly to see if anything fell through the cracks). 🙂 

Another option would be to monitor site traffic and see if you're getting an abundance of traffic from a country or region you wouldn't serve. For example, if you're located in the USA and only serve USA residents, if you're receiving a large amount of traffic from a country like Russia, you could potentially benefit from blocking traffic to your site from Russia, etc.

**Reply by pixelman** [reply]
Thankx kj Martens (sounds like a Flemish name :)
